homeassistant/components/wiz/light.py

```diff
@@ -0,0 +1,348 @@
+                    sceneid,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    def get_min_mireds(self) -> int:
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+        # DW bulbs have no kelvin
+        if self._bulbtype.bulb_type == BulbClass.DW:
+            return 0
+        # If bulbtype is TW or RGB then return the kelvin value
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.max
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.debug("Kelvin is not present in the library. Fallback to 6500")
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    def get_max_mireds(self) -> int:
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+        # DW bulbs have no kelvin
+        if self._bulbtype.bulb_type == BulbClass.DW:
+            return 0
+        # If bulbtype is TW or RGB then return the kelvin value
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.debug("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    def get_supported_features(self) -> int:
+        """Flag supported features."""
+        if not self._bulbtype:
+            # fallback
+            return SUPPORT_FEATURES_RGB
+        features = 0
+        try:
+            # Map features for better reading
+            if self._bulbtype.features.brightness:
+                features = features | SUPPORT_BRIGHTNESS
+            if self._bulbtype.features.color:
+                features = features | SUPPORT_COLOR
+            if self._bulbtype.features.effect:
+                features = features | SUPPORT_EFFECT
+            if self._bulbtype.features.color_tmp:
+                features = features | SUPPORT_COLOR_TEMP
+            return features
+        except WizLightNotKnownBulb:
+            _LOGGER.warning(
```

Comment:
We don't want side effects like logging in the init method where `get_supported_features` is called. Maybe calculate this before creating the entity and pass the result to the entity?

Reply:
#65752